apisix/debug.lua

```diff
@@ -200,7 +202,43 @@ local function sync_debug_status(premature)
 end
 
 
+local function check()
+    if not debug_yaml or not debug_yaml.http then
+        return false
+    end
+
+    local http = debug_yaml.http
+    if not http or not http.enable_header_name or not http.dynamic then
+        return false
+    end
+
+    return true
+end
+
+function _M.dynamic_enable(api_ctx)
```

Comment:
This function is called in the request level, but it changes a field at the module level. Look like it will affect other requests?